packages/mui-base/src/Portal/Portal.tsx

```diff
@@ -41,24 +51,24 @@ const Portal = React.forwardRef(function Portal(props, ref) {
 
   if (disablePortal) {
     if (React.isValidElement(children)) {
-      return React.cloneElement(children, {
-        ref: handleRef,
-      });
+      const newProps = {
+        ref: handleRef
+      }
+      return React.cloneElement(children, newProps);
     }
-    return children;
+    return <React.Fragment>children</React.Fragment>
   }
 
   return (
     <React.Fragment>
-      {mountNode ? ReactDOM.createPortal(children, mountNode) : mountNode}
+      {mountNode ? ReactDOM.createPortal(children, mountNode as HTMLElement) : mountNode}
```

Comment:
The cast should not be needed anymore

Reply:
@akshaya-venkatesh8 please resolve comments only after you addressed them.